Huh

regularnmae opened the pull request:
-Added video generate job, running in the background independet from the browser
-Fixed archive download, when large file (Poorly done but works for centos)
-Disabled missing files check (Filtering events takes forever on utilized disk storage)
-Fixed montagereview player showing NO DATA (Works only with bulk frames on)

regularnmae commented:
> 
> 
> Thank you for taking the time to write this.
> This may conflict with some of the work that @connortechnology has done so we may need to coordinate that.
> 
> For the short term, please fix the hard coded paths in your PR.
> Hard coded paths to files, binaries, and folders must be replaced with an existing ZM_XXXX variable.
> See other areas of the code to determine what those values should be.

Yes, there is hard coded everything not only paths, but I'm not sure anyone need this? Should I continue development or just leave as it is? Because storing frames as images is not efficient way of storing huge amount of data. So I'm concerned about future of this project

knight-of-ni commented:
In order to get others to test your pr, you need to change it so that those not running centos can use it. That means fixing the hard coded paths to ffmpeg, the temp folder, etc. This should be pretty a simple task, compared to the work you have already done. I can't say what we will do with your pr after that. We need to able to test it first.

Sure, jpeg storage is inefficient. No one disputes that, but storage is cheap. We've already incorporated mp4 storage into ZoneMinder, so it should be fairly obvious the direction the project is headed. If you really want to be a big help, send us a PR that resolves the jpeg requirement for some of the auxillary functions in ZoneMinder.

regularnmae commented on web/skins/classic/views/report_event_audit.php:
```diff
@@ -102,12 +102,12 @@
-  } else if ( ! $Event->file_size() ) {
-    $EventsByMonitor[$event['MonitorId']]['ZeroSize'][] = $Event;
-  }
+  } # end if has previous events # DISABLED takes very long time on busy disk sotarage when saving images
```
Agree, and there is a help tip in the zmaudit options

> This is recommended for most systems however if you have a very large number of events the process of scanning the database and filesystem may take a long time and impact performance. In this case you may prefer to not have zmaudit running unconditionally and schedule occasional checks at other, more convenient, times. 

I need to disable zmaudit but not "missing files check" in source, I'm feeling stupid now

> and is the actual reason this view exists.

Well, If I disable audit checks then audit report page shoud disappear, right?

regularnmae replied:
OK I got it, you can run zmaudit manually and use report page to view results, but if zmaudit doing same thing about missing files, than why is nessesary that filtering events checking files additionally? It can be stored in database?

regularnmae commented:
-Fixed hard coded variables except framerate, because can't figure out how to find actual framerate of the monitor
-Reworked archive download
-Removed missing files "fix"
-Removed bulk frames fix, because it should be option?

connortechnology commented on web/ajax/generate.php:
```diff
@@ -12,7 +12,11 @@
   die('Request parameters missing');
 }
 
+require_once( '../includes/config.php' );
```
This shouldn't be necessary as it is included in index.php.  Doesn't hurt to have it though.

connortechnology commented on web/includes/generate_job.php:
```diff
@@ -68,27 +73,29 @@
-  $commandline1 = "cd /mnt/rec/events/" . $event["MonitorId"] . "/" . $date  .  "/" . $eid;
-  $commandline2 = '/usr/bin/ffmpeg -y -framerate ' . $framerate . ' -i %05d-capture.jpg -s ' . $sizeStr . ' ' . 
-    $encoderStr . ' -pix_fmt yuvj420p "Event-_' . $eid . '-r' . $framerateStr  . '-s' . $size  . '.avi"';
+  $commandline1 = 'cd ' . ZM_DIR_EVENTS . '/' . $event['MonitorId'] . '/' . $date  .  '/' . $eid;
```
We probably should have advised better on this.  ZM_DIR_EVENTS used to be the thing to do, but we can have multiple different storage locations.  The Event object has methods that can get you the correct path to the monitor. 
So you can do 
$Event = new Event($event); 
and then do 
= 'cd ' . $Event->Path();

regularnmae replied:
I think better approach would be to use function **createVideo** which generates videos using perl script zmvideo.pl

regularnmae commented:
OMG I REWROTE EVERYTHING, but it works..  and I like it :-P
